Content.Server/Spreader/KudzuSystem.cs

```diff
@@ -83,32 +107,48 @@ private void SetupKudzu(EntityUid uid, KudzuComponent component, ComponentStartu
     /// <inheritdoc/>
     public override void Update(float frameTime)
     {
-        var query = EntityQueryEnumerator<GrowingKudzuComponent, AppearanceComponent>();
+        var query = EntityQueryEnumerator<GrowingKudzuComponent, KudzuComponent, AppearanceComponent>();
         var curTime = _timing.CurTime;
 
-        while (query.MoveNext(out var uid, out var kudzu, out var appearance))
+        while (query.MoveNext(out var uid, out var grow, out var kudzu, out var appearance))
         {
-            if (kudzu.NextTick > curTime)
+            if (grow.NextTick > curTime)
             {
                 continue;
             }
 
-            kudzu.NextTick = curTime + TimeSpan.FromSeconds(0.5);
+            grow.NextTick = curTime + TimeSpan.FromSeconds(0.5);
 
             if (!_robustRandom.Prob(kudzu.GrowthTickChance))
             {
                 continue;
             }
 
-            kudzu.GrowthLevel += 1;
+            if (TryComp<DamageableComponent>(uid, out var damage))
+            {
+                if (damage.TotalDamage > 1.0){
+                    // Gradually heal
+                    _damageable.TryChangeDamage(uid, kudzu.DamageRecovery, true);
+                    if (damage.TotalDamage >= kudzu.GrowthBlock)
+                    {
+                        // Don't grow when quite damaged
+                        if (_robustRandom.Prob(0.95f))
+                        {
+                            continue;
+                        }
+                    }
+                }
+            }
+
+            grow.GrowthLevel += 1;
 
```

Comment:
Considering there's gameplay implications to it now you should remove appearancecomponent from the query because kudzu should still work even without appearance.

Reply:
Done.